### PR TITLE
Treat localhost as a special use case secure domain

### PR DESCRIPTION
**Pull request by saket-ranjan-010**
This change is being made to start treating localhost as a secure special use case domain so that secure cookies are sent to localhost over http connection.

Ref ticket: https://postmanlabs.atlassian.net/browse/RUNTIME-6383

Once this is published we'll bump the tough-cookie versions in the following packages:
- postman-runtime
- postman-request
- postman-sandbox
- postman-app

### CODE REVIEW

File: lib/version.js [hunk 1]
```diff
@@ -1,2 +1,2 @@
 // generated by genversion
-module.exports = '4.1.2-postman.1'
```
Comment (coditva) [inline lib/version.js]: Why do we have this in the diff? 🤔

File: lib/cookie.js [hunk 2]
```diff
@@ -1332,7 +1354,9 @@ class CookieJar {
     if (
       secure == null &&
       context.protocol &&
-      (context.protocol == "https:" || context.protocol == "wss:")
+      (context.protocol == "https:" ||
+        context.protocol == "wss:" ||
+        isSpecialDomain(host))
```
Comment (coditva) [inline lib/cookie.js]: Let's also leave a comment with reference to the RFC about why this is check is being made.

**Review by coditva — COMMENTED** [review]
@saket-ranjan-010 Let's add reference to the RFC for the change.
Can we add a test for this behaviour (which breaks without this change)?

Also, we might want to send this change upstream once this PR is merged.

File: lib/version.js (hunk 1 above)
Comment (saket-ranjan-010) [reply lib/version.js]: @coditva The master pipeline was failing after the last release. This fixes that failure.

File: lib/cookie.js (hunk 2 above)
Comment (saket-ranjan-010) [reply lib/cookie.js]: Done.

**Review by codenirvana — APPROVED** [review]

**Pull request merged**